src/app/core/views/Preferences/tabs/Billing/CancelSubscriptionModal.tsx

```diff
@@ -24,30 +28,60 @@ const CancelSubscriptionModal = ({
   cancelSubscription: (feedback: string) => void;
 }): JSX.Element => {
   const { translate } = useTranslationContext();
-  const [step, setStep] = useState<1 | 2>(1);
+  const [step, setStep] = useState<1 | 2 | 3>(2);
+  const [couponAvailable, setCouponAvailable] = useState(false);
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
-    if (isOpen) {
-      setStep(1);
+    paymentService.getCoupon().then((coupon) => setCouponAvailable(coupon.elegible));
+  }, []);
+
+  useEffect(() => {
+    couponAvailable && setStep(1);
+  }, [couponAvailable]);
+
+  const applyCoupon = async () => {
+    try {
+      await paymentService.applyCoupon();
+      notificationsService.show({ text: translate('notificationMessages.successApplyCoupon') });
+    } catch (error) {
+      console.error(error);
+      notificationsService.show({
```

Comment:
What if the coupon has been already applied and the user is not eligible? Do we show something more specific? If we are working on reducing the churn, this could happen a lot, even if we handle the modal not being shown. (p.e: a user that left a window before applying the coupon opened with the modal, a lot of people could try to trick us). 

So, if someone tries something tricky or any bug is introduced in the future on the front end, showing that the user is not eligible will be enough to avoid unnecessary tickets of people complaining about the coupon appliance failing.

Reply:
If the coupon has been applied, the modal is not displayed, but I am not sure if I understood you.
Do you want to change the message that is displayed in case the coupon cannot be applied?

Reply:
So, assume that I open a window and the modal appears, but I left the web app there, without pressing the apply button. Then I open another window and the modal appears again - as I have not still the coupon applied - and then I apply it. After doing that, I return to the first window and press the apply button. I am going to get a generic error. 

This is a corner case and is not too important, but some users are going to try that. 

Also, even ignoring that case, if we introduce some bug unintentionally and the modal shows when should not, we should still show the proper error message, which in that case would be 'The coupon has been already applied' or something like this.